op-service/util.go

```diff
@@ -170,3 +174,48 @@ func U64UtilPtr(in uint64) *hexutil.Uint64 {
 	util := hexutil.Uint64(in)
 	return &util
 }
+
+// this is a wrapper over ethclient to fix certain methods getting wrongly exposed by ethclient
+type L1Client struct {
+	l1Rpc             *rpc.Client
+	*ethclient.Client // fallback for unimplemented methods
+}
+
+func NewL1Client(l1Rpc *rpc.Client) *L1Client {
+	return &L1Client{
+		l1Rpc:  l1Rpc,
+		Client: ethclient.NewClient(l1Rpc),
+	}
+}
+
+func DialL1Client(l1RpcURL string) (*L1Client, error) {
+	if l1RpcURL == "" {
+		return nil, fmt.Errorf("l1 Rpc Url is required")
+	}
+	l1Rpc, err := rpc.Dial(l1RpcURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to L1 RPC: %w", err)
+	}
+	return NewL1Client(l1Rpc), nil
+}
+
+// not using using genesis.BlockRef to avoid a semantically looking cyclic dependency
```

Comment:
@mslipper I saw a bunch of imports of op-service in genesis pkg, so didn't want to use the BlockRef defined in there to use here to create a cyclic-looking dependency.